DOCS-#3040: document `modin.backends.pandas` (#3041)
modin-project/modin: docs/flow/modin/backends/pandas/query_compiler.rst

Pandas query compiler
"""""""""""""""""""""

> `PandasQueryCompiler` is responsible for compiling efficient DataFrame algebra queries for the

Reviewer: ```suggestion
``PandasQueryCompiler`` is responsible for compiling a set of known user-defined functions
and pairing those with dataframe algebra operators in the
```

Author: What is "a set of known UDFs" and how does pandas query compiler responsible for compiling those?

Reviewer: We use pandas functions as the UDFs usually, but sometimes we construct different functions. Since the backend does not assume anything about the functions they can be arbitrary. Maybe instead of user-defined functions we can use something like developer defined functions or predefined functions.

Author: changed to:
"``PandasQueryCompiler`` is responsible for compiling a set of known **predefined** functions and pairing those with dataframe algebra operators"

> :doc:`BasePandasFrame </flow/modin/engines/base/frame/data>`, the such frame that contains

Reviewer: ```suggestion
:doc:`BasePandasFrame </flow/modin/engines/base/frame/data>`, specifically for dataframes backed by
```

> pandas DataFrames in its partitions as a payload.

Reviewer: ```suggestion
``pandas.DataFrame`` objects.
```

Each `PandasQueryCompiler` contains an instance of `BasePandasFrame` which it queries to get the result.

> `PandasQueryCompiler` supports methods built by :doc:`function module </flow/modin/data_management/functions>`,

Reviewer: ```suggestion
`PandasQueryCompiler` supports methods built by the :doc:`function module </flow/modin/data_management/functions>`.
```

> so if you want to add implementation for some of the query compilers API method you may visit this module

Reviewer: ```suggestion
If you want to add an implementation for a query compiler method, visit the function module documentation
```

> to inspect whether the new operation suits one of the presented templates and can be easily implemented

Reviewer: ```suggestion
to see whether the new operation fits one of the existing function templates and can be easily implemented
```
